pkg/chunk/series_store.go

```diff
@@ -252,10 +262,21 @@ func (c *seriesStore) lookupSeriesByMetricNameMatchers(ctx context.Context, from
 	log, ctx := spanlogger.New(ctx, "SeriesStore.lookupSeriesByMetricNameMatchers", "metricName", metricName, "matchers", len(matchers))
 	defer log.Span.Finish()
 
+	// Check if one of the labels is a shard annotation, pass that information to lookupSeriesByMetricNameMatcher,
+	// and remove the label.
```

Comment:
We need to remove the matcher because it doesn't exist in the store. if we leave it `lookupSeriesByMetricNameMatcher` would returns no result. Basically the matchers is used to convey the information until here.